### PR TITLE
Created Burger Button

### PR DESCRIPTION
**Pull request by CressChe**
added Burger button + menu + tests

### CODE REVIEW

File: src/components/NavBar.tsx [hunk 1]
```diff
@@ -1,15 +1,31 @@
 import React from 'react';
-import { Link } from 'react-router-dom';
-import '../styles/NavBar.scss'
+import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
+import '../styles/BurgerButton.scss'
+import { useState } from 'react';
+import '../styles/NavBar.scss';
+
```
Comment (oskarwilliams) [inline src/components/NavBar.tsx:5]: Is this new line needed?

File: src/components/NavBar.tsx [hunk 2]
```diff
@@ -1,15 +1,31 @@
 import React from 'react';
-import { Link } from 'react-router-dom';
-import '../styles/NavBar.scss'
+import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
```
Comment (oskarwilliams) [inline src/components/NavBar.tsx]: Do we need to import all of these?

File: src/components/NavBar.tsx [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
+                <Link to='/spirit' className = 'nav-link'>Spirit </Link>
+                <Link to='/opportunity'className = 'nav-link'>Opportunity</Link>
+                <Link to='/curiosity'className = 'nav-link'>Curiosity</Link>
+                <Link to='/timeline'className = 'nav-link' > Timeline</Link>
+            </div>
+
+            <div className="burger-button" 
+                onClick={() => {setBurgerState(!closeBurger)}
+            } data-testid= 'burger-test'>
+                <div className={closeBurger ? "opened-bar-1" : "closed-bar-1"}
```
Comment (oskarwilliams) [inline src/components/NavBar.tsx:23]: is there a reason you decided to not have the open close on the parent element?

File: src/components/NavBar.tsx [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
+                <Link to='/' className = 'nav-link'> Home </Link>
+                <Link to='/spirit' className = 'nav-link'>Spirit </Link>
+                <Link to='/opportunity'className = 'nav-link'>Opportunity</Link>
+                <Link to='/curiosity'className = 'nav-link'>Curiosity</Link>
+                <Link to='/timeline'className = 'nav-link' > Timeline</Link>
+            </div>
+
+            <div className="burger-button" 
+                onClick={() => {setBurgerState(!closeBurger)}
+            } data-testid= 'burger-test'>
```
Comment (oskarwilliams) [inline src/components/NavBar.tsx]:
```suggestion
                onClick={() => {setBurgerState(!closeBurger)}}
                data-testid= 'burger-test'>
```

File: src/styles/BurgerButton.scss [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  }
+  
+  /* Rotate last bar */
+  .closed-bar-3 {
+    @include bar;
+    -webkit-transform: rotate(45deg) translate(-8px, -8px) ;
+    transform: rotate(45deg) translate(-8px, -8px) ;
+  }
+
+  @media(min-width: $tabletScreenSize) {
```
Comment (oskarwilliams) [inline src/styles/BurgerButton.scss]: I would probably do this instead, we don't need a variable for a size that is 1px larger than another variable. Also if $mobileScreenSize is the max screen size of a mobile, I would expect $tabletScreenSize to be the max screen size of the tablet, which isn't what this represents here
```suggestion
  @media(min-width: $mobileScreenSize + 1px) {
```

File: src/tests/NavBar.test.tsx [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, shallow } from '@testing-library/react';
```
Comment (oskarwilliams) [inline src/tests/NavBar.test.tsx]: you can remove shallow from here :)

**Review by oskarwilliams — CHANGES_REQUESTED** [review]

File: src/components/NavBar.tsx (hunk 3 above)
Comment (CressChe) [reply src/components/NavBar.tsx:23]: We couldn't get the other way to work - burger button disappeared :(

**Review by oskarwilliams — APPROVED** [review]

**Pull request merged**